src/server/game/Handlers/CalendarHandler.cpp

```diff
@@ -563,61 +411,45 @@ void WorldSession::HandleCalendarEventRsvp(WorldPacket& recvData)
         "], ownerInviteId [" UI64FMTD "], Invitee ([%s] id: [" UI64FMTD "])",
-        guid.ToString().c_str(), eventId, ownerInviteId, invitee.ToString().c_str(), inviteId);
+        guid.ToString().c_str(), calendarRemoveInvite.EventID, calendarRemoveInvite.ModeratorID, calendarRemoveInvite.Guid.ToString().c_str(), calendarRemoveInvite.InviteID);
 
-    if (CalendarEvent* calendarEvent = sCalendarMgr->GetEvent(eventId))
+    if (CalendarEvent* calendarEvent = sCalendarMgr->GetEvent(calendarRemoveInvite.EventID))
     {
-        if (calendarEvent->GetCreatorGUID() == invitee)
+        if (calendarEvent->GetOwnerGUID() == calendarRemoveInvite.Guid)
         {
             sCalendarMgr->SendCalendarCommandResult(guid, CALENDAR_ERROR_DELETE_CREATOR_FAILED);
             return;
         }
 
-        sCalendarMgr->RemoveInvite(inviteId, eventId, guid);
+        sCalendarMgr->RemoveInvite(calendarRemoveInvite.InviteID, calendarRemoveInvite.EventID, guid);
     }
     else
         sCalendarMgr->SendCalendarCommandResult(guid, CALENDAR_ERROR_NO_INVITE);
 }
 
-void WorldSession::HandleCalendarEventStatus(WorldPacket& recvData)
+void WorldSession::HandleCalendarEventStatus(WorldPackets::Calendar::CalendarEventStatus& calendarEventStatus)
 {
     ObjectGuid guid = _player->GetGUID();
-    ObjectGuid invitee;
-    uint64 eventId;
-    uint64 inviteId;
-    uint64 ownerInviteId; // isn't it sender's inviteId?
-    uint8 status;
 
-    recvData >> invitee.ReadAsPacked();
-    recvData >> eventId >> inviteId >> ownerInviteId >> status;
     TC_LOG_DEBUG("network", "CMSG_CALENDAR_EVENT_STATUS [%s] EventId ["
         UI64FMTD "] ownerInviteId [" UI64FMTD "], Invitee ([%s] id: ["
-        UI64FMTD "], status %u", guid.ToString().c_str(), eventId, ownerInviteId, invitee.ToString().c_str(), inviteId, status);
+        UI64FMTD "], status %u", guid.ToString().c_str(), calendarEventStatus.EventID, calendarEventStatus.ModeratorID, calendarEventStatus.Guid.ToString().c_str(), calendarEventStatus.InviteID, calendarEventStatus.Status);
 
-    if (CalendarEvent* calendarEvent = sCalendarMgr->GetEvent(eventId))
+    if (CalendarEvent* calendarEvent = sCalendarMgr->GetEvent(calendarEventStatus.EventID))
     {
-        if (CalendarInvite* invite = sCalendarMgr->GetInvite(inviteId))
+        if (CalendarInvite* invite = sCalendarMgr->GetInvite(calendarEventStatus.InviteID))
         {
-            invite->SetStatus((CalendarInviteStatus)status);
-            // not sure if we should set response time when moderator changes invite status
-            //invite->SetStatusTime(time(NULL));
+            invite->SetStatus((CalendarInviteStatus)calendarEventStatus.Status);
 
             sCalendarMgr->UpdateInvite(invite);
             sCalendarMgr->SendCalendarEventStatus(*calendarEvent, *invite);
-            sCalendarMgr->SendCalendarClearPendingAction(invitee);
+            //sCalendarMgr->SendCalendarClearPendingAction(calendarEventStatus.Guid);
```

Comment:
Not sure if SMSG_CALENDAR_CLEAR_PENDING_ACTION should be called now. But I haven't really looked at it, so I will remove this change :)